### PR TITLE
docs: add forms guide

### PR DESCRIPTION
**Pull request by zettca**
- add Forms guide section from v4
- add vanilla form example & docs
- improve samples

### CODE REVIEW

File: docs/guides/forms/Forms.stories.mdx [hunk 1]
```diff
@@ -1,5 +1,79 @@
...
+
+## Form Components
+
+UI Kit provides a set of **Form Components** to build forms. These components have the goals of providing a **consistent React API** to build forms **following the HV Design spec** and
+**ease of use and integration with** native HTML form, React forms and form libraries.
+
+The base API of all the form components is established by the **[Form Element](https://lumada-design.github.io/uikit/master/?path=/docs/guides-forms-form-element--docs)** component.
+It provides the necessary props to listen to the events, manage the component `status` as well as a `value` prop to make each component **controlled**.
+
+The UI Kit library currently offers the following form components:
```
Comment (MEsteves22) [inline docs/guides/forms/Forms.stories.mdx:19]: I think the Filter Group is missing as a form component if I'm not mistaken.

File: docs/guides/forms/Forms.stories.mdx [hunk 2]
```diff
@@ -1,5 +1,79 @@
...
+import { Canvas, Meta, Story } from "@storybook/addon-docs";
+import * as FormSamples from "./FormsSamples";
 
-<Meta title="Guides/Forms/Main" />
+<Meta title="Guides/Forms/Guide" />
 
 # Forms
+
+Web forms play a crucial role in web applications, as they are the primary form to collect user information.
+By providing a standardised way to capture user input and validat it, we can ensure a consistent user experience across the different applications.
```
Comment (MEsteves22) [inline docs/guides/forms/Forms.stories.mdx:9]: Typo: "validat" => "validate"

File: docs/guides/forms/Forms.stories.mdx [hunk 3]
```diff
@@ -1,5 +1,79 @@
...
+and reading the results from `FormData`.
+It's a good option for **simple forms** using form components based on native form controls (eg. `input`) where **custom validation is not needed**.
+
+**Note:** Not all UI Kit form components integrate with the native form data or validation.
+
+<Canvas>
+  <Story name="Native Form" story={FormSamples.NativeFormStory} />
+</Canvas>
+
+### React form
```
Comment (MEsteves22) [inline docs/guides/forms/Forms.stories.mdx:50]: Maybe we could make the text field and dropdown fields required for the native, react, react hook, and formik forms just to showcase some error messages

**Review by MEsteves22 — COMMENTED** [review]
Great docs 🙏 

I just left some minor comments:

File: docs/guides/forms/Forms.stories.mdx (hunk 1 above)
Comment (zettca) [reply docs/guides/forms/Forms.stories.mdx:19]: It's not - I'll add a few words about it 👍

Comment (MEsteves22) [reply docs/guides/forms/Forms.stories.mdx:19]: Okay, my bad! I though it was since it uses the `HvFormElement` 😅

**Review by plagoa — APPROVED** [review]

**Review by MEsteves22 — APPROVED** [review]

**Comment by MEsteves22** [comment]
Applitools is never able to run 😅 It seems that `protocolTimeout` to 5 minutes was not enough. Should we increase the timeout?

**Review by plagoa — APPROVED** [review]

**Pull request merged**